### PR TITLE
refactor: automatically approve comments or replies after admin replied it

### LINKED ISSUE #5870: 未审核通过的评论管理员可以回复
### 系统信息

2.15.1

### 使用的哪种方式运行？

Docker Compose

### 发生了什么？

未审核通过的评论管理员可以回复，应该不可以回复吧，或者回复了直接置为审核通过

### 相关日志输出

_No response_

### 附加信息

_No response_

### PR DESCRIPTION
**Pull request by guqing**
#### What type of PR is this?
/kind improvement
/area core
/milestone 2.16.x

#### What this PR does / why we need it:
管理员回复评论或回复后自动通过审核

#### Which issue(s) this PR fixes:
Fixes #5870

#### Does this PR introduce a user-facing change?
```release-note
管理员回复评论或回复后自动通过审核
```

### CODE REVIEW

**Comment by guqing** [comment]
@ruibaby Console 回复评论后点击展开会因为乐观锁出现 500 错误，原因展开评论的回复会标记最后已读时间可能需要加上重试

File: application/src/main/java/run/halo/app/content/comment/ReplyServiceImpl.java [hunk 1]
```diff
@@ -54,56 +57,100 @@ public class ReplyServiceImpl implements ReplyService {
...
+                        .thenReturn(createdReply);
+                })
+                .flatMap(createdReply -> approveComment(comment)
+                    .thenReturn(createdReply)
+                )
+            );
+    }
+
+    private Mono<Comment> approveComment(Comment comment) {
+        UnaryOperator<Comment> updateFunc = commentToUpdate -> {
```
Comment (JohnNiang) [inline application/src/main/java/run/halo/app/content/comment/ReplyServiceImpl.java:77]: 这里推荐用 `Mono.just(comment).doOnNext(c -> c.setxxx ...).flatMap(client::update)`.

Comment (JohnNiang) [reply application/src/main/java/run/halo/app/content/comment/ReplyServiceImpl.java:77]: 我忽略了 updateFunc 在 updateCommentWithRetry 会用到。请忽略我。

**Review by JohnNiang — APPROVED** [review]
/approve

**Review by ruibaby — APPROVED** [review]
/lgtm

**Pull request merged**